### PR TITLE
Get CM trigger data

### PR DESCRIPTION
**Pull request by piotrkonowrocki**
# Get CM trigger data

Also covers parts from https://app.shortcut.com/oazo-apps/story/5220/constant-multiple-information-section.
  
## Changes 👷‍♀️

- Updated contract address and abi according to newest version,
- loaded CM trigger into code and displayed it into details section,
- extended `extractBasicBSData` method with `isInGroup` parameter and changed its props to object with interface for better readability.
  
## How to test 🧪

- May not be possible to test right now, because cache bot address is not updated and doesn't return trigger after it's created.

### CODE REVIEW

File: blockchain/abi/dummy-automation-bot-aggregator.json [hunk 1]
```diff
@@ -1,390 +1,91 @@
 [
```
Comment (piekczyk) [inline blockchain/abi/dummy-automation-bot-aggregator.json:1]: not sure if it still dummy

File: components/vault/OptimizationControl.tsx [hunk 2]
```diff
@@ -110,7 +110,10 @@ export function OptimizationControl({
...
 
   const basicBuyData = automationTriggersData
-    ? extractBasicBSData(automationTriggersData, TriggerType.BasicBuy)
+    ? extractBasicBSData({
+        triggersData: automationTriggersData,
+        triggerType: TriggerType.BasicBuy,
+      })
     : undefined
 
   const vaultHasActiveTrigger = basicBuyData?.isTriggerEnabled
```
Comment (piekczyk) [inline components/vault/OptimizationControl.tsx:119]: maybe not for now but it should be extended with check whether constant multiple is enabled

File: features/automation/common/basicBSTriggerData.ts [hunk 3]
```diff
@@ -100,14 +106,25 @@ export function extractGroupTriggersData(
   }
 
   return {
-    [TriggerType.CMBasicBuy]: extractBasicBSData(groupData, TriggerType.BasicBuy),
-    [TriggerType.CMBasicSell]: extractBasicBSData(groupData, TriggerType.BasicSell),
+    [TriggerType.CMBasicBuy]: extractBasicBSData({
```
Comment (piekczyk) [inline features/automation/common/basicBSTriggerData.ts:109]: we no longer have CMBasicBuy or CmBasicSell trigger type, we should operate on BasicBuy / BasicSell (automation package to be updated I guess)

File: features/automation/common/basicBSTriggerData.ts [hunk 4]
```diff
@@ -100,14 +106,25 @@ export function extractGroupTriggersData(
...
+      triggersData: groupData,
+      triggerType: TriggerType.BasicBuy,
```
Comment (piekczyk) [inline features/automation/common/basicBSTriggerData.ts:111]: shouldn't we pass here and below `isInGroup` as `true`?

File: features/automation/common/basicBSTriggerData.ts [hunk 5]
```diff
@@ -100,14 +106,25 @@ export function extractGroupTriggersData(
...
+    }),
+    [TriggerType.CMBasicSell]: extractBasicBSData({
+      triggersData: groupData,
+      triggerType: TriggerType.BasicSell,
+    }),
   }
 }
```
Comment (piekczyk) [inline features/automation/common/basicBSTriggerData.ts:118]: ok, I guess this whole method is not used everywhere. If its true lets remove it.

**Review by piekczyk — CHANGES_REQUESTED** [review]

File: components/vault/OptimizationControl.tsx (hunk 2 above)
Comment (piotrkonowrocki) [reply components/vault/OptimizationControl.tsx:119]: I plan to check all CM logic for things like that while working on default state, this PR is just about getting data so I can move on with next thing.

File: features/automation/common/basicBSTriggerData.ts (hunk 3 above)
Comment (piotrkonowrocki) [reply features/automation/common/basicBSTriggerData.ts:109]: This method is used in `useConstantMultipleStateInitialization` which I intend to refactor and remove some old stuff with default state.

File: features/automation/common/basicBSTriggerData.ts (hunk 4 above)
Comment (piotrkonowrocki) [reply features/automation/common/basicBSTriggerData.ts:111]: Same thing, this method is probably going to be removed, the only reason I updated is that it was required to work in current state and parameters format changed.

File: features/automation/common/basicBSTriggerData.ts (hunk 5 above)
Comment (piotrkonowrocki) [reply features/automation/common/basicBSTriggerData.ts:118]: Currently is used in `useConstantMultipleStateInitialization`, will be removed later.

File: blockchain/abi/dummy-automation-bot-aggregator.json (hunk 1 above)
Comment (piotrkonowrocki) [reply blockchain/abi/dummy-automation-bot-aggregator.json]: Renamed, removed `dummy`.

**Review by piekczyk — APPROVED** [review]

**Pull request merged**